### ISSUE TITLE
BiocManager::repositories() Reports Missing `books` for 3.15/3.16

### ISSUE THREAD

**Issue by jonyoder** [Bioconductor/BiocManager#135]
When using Bioconductor 3.15 or 3.16, the command `BiocManager::repositories()` reports that the `book` repository is available, but the repository does not appear to exist on bioconductor.org.

**Comment by lshep** [comment]
Do you get an ERROR when trying to access a book? The books are handle a bit differently and do not appear on the [biocViews page](http://bioconductor.org/packages/release/BiocViews.html#___Software) .  You can access their landing page, which is a rendering of the book, through the [build report page]( http://bioconductor.org/checkResults/).  Example [devel books report](http://bioconductor.org/checkResults/3.16/books-LATEST/) which has "landing page" links

**Comment by jonyoder** [comment]
Thanks for that clarification! I'll close this issue.

**Comment by jonyoder** [comment]
Would it be reasonable to have `BiocManager::repositories()` respect what is currently enabled in `config.yaml`? I see it's currently hard-coded at https://github.com/Bioconductor/BiocManager/blob/master/R/repositories.R#L104.

**Comment by LiNk-NY** [comment]
> Oh, I see what was tripping us up. The `books` repo is missing from `devel_repos` in `config.yaml`. We parse `config.yaml` to figure out what repos are available.

Hi Jonathan, @jonyoder 
 
Lori can correct me if I'm wrong but this field in `config.yaml` is used during the release process. I don't recommend using it as the basis for available repositories. I would refer you to `BiocManager::repositories()` for that rather than parsing the `config.yaml` file.

Best regards,
Marcel

**Comment by lshep** [comment]
yes it is used in the release process as well as some of the BBS code. I agree with marcel's comment above you should use the provided functions rather than he config.yaml.  This seems trivial request but it could actually have many foreseen consequences.  We can look into it but its not as simple as it seems.

**Comment by LiNk-NY** [comment]
Hi Jonathan, @jonyoder 

Thanks for providing some background. That is helpful. 
You're right, we should be using the yaml file as the basis for the `BiocManager::repositories` functions and others. For practical reasons, we recommend using `BiocManager` functions and handling warnings about missing repositories in the codebase. 
We will consider the former approach in future versions of `BiocManager` though it will take time and effort to implement. 

Best regards,
Marcel
